src/app/store/slices/storage/storage.thunks/uploadFolderThunk.ts

```diff
@@ -374,11 +393,36 @@ export const uploadMultipleFolderThunkNoCheck = createAsyncThunk<
             subtitle: t('tasks.subtitles.upload-failed') as string,
           },
         });
-        // Log the error or report it but don't re-throw it to allow the next folder to be processed
         errorService.reportError(castedError);
-        continue;
       }
-    }
+    };
+
+    const checkCompletion = async () => {
+      if (folderQueue.idle() && filesQueue.idle()) {
+        tasksService.updateTask({
+          taskId: taskId,
+          merge: {
+            itemUUID: { rootFolderUUID: rootFolderData?.uuid },
+            status: TaskStatus.Success,
+          },
+        });
+        options.onSuccess?.();
+
+        setTimeout(() => {
+          dispatch(planThunks.fetchUsageThunk());
+          if (memberId) dispatch(planThunks.fetchBusinessLimitUsageThunk());
+        }, 1000);
+
+        dispatch(fetchSortedFolderContentThunk(currentFolderId));
+      }
+    };
+
+    folderQueue.drain(checkCompletion);
+    filesQueue.drain(checkCompletion);
+
+    levels.forEach(async (level) => {
```

Comment:
why `async` without `await`? `forEach` is sequential, so unless you are planning to do a `Promise.all` or something like that, this is not needed